src/rustdoc-json-types/lib.rs

```diff
@@ -390,7 +392,12 @@ pub enum AssocItemConstraintKind {
 /// to parse them, or otherwise depend on any implementation details.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
 // FIXME(aDotInTheVoid): Consider making this non-public in rustdoc-types.
-pub struct Id(pub u32);
+pub struct ItemId(pub u32);
+
+/// A type identifier.
+///
+/// It can be used to lookup in [`Crate::types`] to resolve it to [`Type`].
+pub type TypeId = usize;
```

Comment:
`u32` would be better. There's no risk of overflowing; rustc uses `u32` for code locations which means it can't handle more than 4 GiB of source code.